address.go

```diff
@@ -48,46 +55,48 @@ func newAddressGeneratorAtState(chainID ChainID, state addressState) *AddressGen
 	}
 }
 
-// NextAddress generates an account address and increments
-// the addressing state.
-//
-// The address is generated for a specific network (Flow mainnet, testnet, etc)
-// Each state is mapped to exactly one address. There are as many addresses
-// as states.
-//
-// zeroAddress() corresponds to the state "0" while ServiceAddress() corresponds to the
-// state "1".
-func (gen *AddressGenerator) NextAddress() (Address, error) {
-	err := gen.nextState()
-	if err != nil {
-		return zeroAddress(gen.chainID), err
-	}
+// Address generates an account address at the current addressing state.
+func (gen *AddressGenerator) Address() Address {
+	return generateAddress(gen.chainID, gen.state)
+}
 
-	return generateAddress(gen.chainID, gen.state), nil
+// NextAddress increments the addressing state and generates an account address.
+func (gen *AddressGenerator) NextAddress() Address {
+	gen.Next()
+	return generateAddress(gen.chainID, gen.state)
 }
 
-// nextState increments the addressing state.
+// Next increments the addressing state.
 //
-// The state values are incremented from 0 to 2^k-1
-func (gen *AddressGenerator) nextState() error {
+// State values are incremented from 0 to 2^k-1.
+func (gen *AddressGenerator) Next() *AddressGenerator {
```

Comment:
Just pointing out that having `Next` and `Address` together can replace a call to `NextAddress`. Maybe we can keep only one option to developers not to get them confused?

Reply:
I'm going to keep `NextAddress` in order to stay backwards compatible for the next few releases, but I agree that it should be deprecated in the future.